### ISSUE TITLE
Missing Mask_Predict Generation Strategy

### ISSUE THREAD

**Issue by risingdhxs** [pytorch/fairseq#2653]
## 🚀 Feature Request
Request to have the Mask-Predict generation/inference strategy (proposed in the Mask-Predict paper) available in fairseq

### Motivation

First of all thank you for such a great library. I'm currently poking around the CMLM model proposed in the Mask-Predict paper, and have been using the script [here](https://github.com/pytorch/fairseq/blob/master/examples/nonautoregressive_translation/scripts.md) to train the models. For inference and BLEU calculation I'm using the scripts outlined in the Translate section [here](https://github.com/pytorch/fairseq/blob/master/examples/nonautoregressive_translation/README.md), but that is not using the Mask-Predict algorithm proposed in the paper right? So far I have not been able to find this Inference strategy in the current library, unless I missed something huge?

### Pitch

I'm looking at the Mask-Predict scripts in the paper's repo [here](https://github.com/facebookresearch/Mask-Predict), and trying to make it work with fairseq. It would be ideal to have it merged with fairseq's main branch.
